src/platform/web/lib/http/request/fetchrequest.ts

```diff
@@ -54,6 +54,7 @@ export default async function fetchRequest(
     method: _method,
     headers: fetchHeaders,
     body: body as any,
+    signal: controller.signal,
```

Comment:
What is the point of `controller`? We don't ever call `abort()` on it so I’m not sure what it's achieving.

Ways in which it might be sensible to use `abort()`:

- to cancel in-flight requests when, for example, the underlying `Realtime` instance is closed — I’m not sure if we do this for any of our other HTTP request implementations though
- to provide users with a way to explicitly cancel an in-flight request — we have no API for this

Reply:
Oops, I’m talking nonsense — we do call `abort()` on it, sorry!

Reply:
Another thing, from [here]():

> When `abort()` is called, the `fetch()` promise rejects with a `DOMException` named `AbortError`.

Just wanted to check that our `resolve({ error: new PartialErrorInfo('Request timed out', null, 408) });` beats the `AbortError` in becoming the error that is exposed to the user? Looks like it should, might be worth putting a comment alongside it though.

Reply:
> Oops, I’m talking nonsense — we do call abort() on it, sorry!
Correct! That was the reason behind the issue, we did call abort() before, but never passed the controller signal, so it did nothing.

Reply:
> Just wanted to check that our resolve({ error: new PartialErrorInfo('Request timed out', null, 408) }); beats the AbortError in becoming the error that is exposed to the user? Looks like it should, might be worth putting a comment alongside it though.

I tested it locally before, `resolve({ error: new PartialErrorInfo('Request timed out', null, 408) });` resolves first so we get our formatted error as expected. Will add a comment for that, thanks!

Reply:
Added comment in https://github.com/ably/ably-js/compare/ca0b1db206eaa758fd1ab45a4aa0253a6f6583ec..d418952a56cfe168e515ab9455661991c6456939